cfpb/consumerfinance.gov
# Wait for API response in email signup functional tests

Wait for the stubbed email signup API to return a response before making any assertions. This should prevent any races where Cypress is trying to look for something that will only show up after the API response has been received.

---

## Changes

- Wait for the stubbed email API response before any assertions run

## How to test this PR

1. `yarn cypress open`, run the `email-signup.cy.js` spec
2. Note that Cypress is now waiting for the `subscriptionSuccess` API to return before running the assertions (step 8)

## Screenshots

![wait](https://user-images.githubusercontent.com/1862695/195912688-7b337349-4223-414f-b41b-4c6670acac1c.png)

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
